t/plugin/grpc-transcode4.t

```diff
@@ -0,0 +1,154 @@
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+use t::APISIX 'no_plan';
+# ensure that the JSON module of Perl is installed in your test environment.
+# If it is not installed, sudo cpanm JSON.
+use JSON;
+
+no_long_string();
+no_shuffle();
+no_root_location();
+
+add_block_preprocessor(sub {
+    my ($block) = @_;
+
+    if (!$block->request) {
+        $block->set_value("request", "GET /t");
+    }
+});
+
+run_tests;
+
+__DATA__
+
+=== TEST 1: set rule
+--- config
+    location /t {
+       content_by_lua_block {
+          local http = require "resty.http"
+          local t = require("lib.test_admin").test
+          local code, body = t('/apisix/admin/protos/1',
+                ngx.HTTP_PUT,
+                [[{
+                   "content" : "syntax = \"proto3\";
+                    package user;
+                    service UserService {
+                        rpc GetUserInfo(UserRequest) returns (UserResponse) {}
+                     }
+
+                    enum Gender {
+                        GENDER_UNSPECIFIED = 0;
+                        GENDER_MALE = 1;
+                        GENDER_FEMALE = 2;
+                    }
+                    message Job {
+                        string items = 1;
+                    }
+                    message UserRequest {
+                        string name = 1;
+                        int32 age = 2;
+                    }
+
+                    message UserResponse {
+                        Gender gender = 1;
+                        repeated string items = 2;
+                        string message = 3;
+                        Job job = 4;
+                    }"
+                }]]
+              )
+
+             if code >= 300 then
+                 ngx.say(body)
+                 return
+              end
+
+             local code, body = t('/apisix/admin/routes/1',
+                ngx.HTTP_PUT,
+                [[{
+                   "methods": ["POST"],
+                    "uri": "/grpctest",
+                    "plugins": {
+                        "grpc-transcode": {
+                            "proto_id": "1",
+                            "service": "user.UserService",
+                            "method": "GetUserInfo"
+                       }
+                    },
+                    "upstream": {
+                        "scheme": "grpc",
+                            "type": "roundrobin",
+                            "nodes": {
+                            "127.0.0.1:50051": 1
```

Comment:
I can't found this test backend in APISIX e2e framework, I think this new added test cased can't be pass due to https://github.com/apache/apisix/pull/12462  ,  try to merge master branch and re-run CI again.